src/libclipper/src/datatypes.cpp

```diff
@@ -268,6 +269,27 @@ std::vector<ByteBuffer> rpc::PredictionRequest::serialize() {
   return serialized_request;
 }
 
+rpc::PredictionResponse::PredictionResponse(
+    const std::vector<std::string> outputs)
+    : outputs_(outputs) {}
+
+rpc::PredictionResponse
+rpc::PredictionResponse::deserialize_prediction_response(ByteBuffer bytes) {
+  std::vector<std::string> outputs;
+  uint32_t *output_lengths_data = reinterpret_cast<uint32_t *>(bytes.data());
+  uint32_t num_outputs = output_lengths_data[0];
+  output_lengths_data++;
+  char *output_string_data = reinterpret_cast<char *>(
+      bytes.data() + sizeof(uint32_t) + (num_outputs * sizeof(uint32_t)));
+  for (uint32_t i = 0; i < num_outputs; i++) {
+    uint32_t output_length = output_lengths_data[i];
+    std::string output(output_string_data, output_length);
+    outputs.push_back(std::move(output));
+    output_string_data += output_length;
+  }
+  return PredictionResponse(outputs);
+}
```

Comment:
Added one! Renamed `input_test` to `data_serialization_test` and added `data_serialization_test#PredictionResponseDeserialization`.